cleverhans/attacks.py

```diff
@@ -1821,3 +1821,8 @@ def loss_fn(x, label):
             is_debug=is_debug,
         )
         return adv_x
+
+    def generate_np(self, x_val, **kwargs):
+        # Add shape check for batch size=1, then call parent class generate_np
+        assert x_val.shape[0] == 1, 'x_val should be a batch of a single image'
```

Comment:
Can we allow arbitrary batch sizes and then iterate through them in the function here? It seems like the "x_val must be a batch of a single image" constraint will probably be surprising to people.